contactsummary/js/src/contact-summary/services/contract.service.js

```diff
@@ -94,32 +114,12 @@ define([
 
     function getContracts () {
       var deferred = $q.defer();
-      if (_.isEmpty(contracts)) {
-        ContactDetails.get()
-          .then(function (response) {
-            var data = {
-              contact_id: response.id,
-              'api.HRJobContractRevision.getcurrentrevision': {jobcontract_id: '$value.id'}
-            };
-
-            return Api.get('HRJobContract', data);
-          })
-          .then(function (response) {
-            var activeContracts = response.values.filter(function (contract) {
-              return parseInt(contract.deleted) === 0;
-            });
 
-            if (activeContracts.length === 0) {
-              return deferred.reject('No job contract found');
-            }
-
-            contracts = activeContracts;
-
-            deferred.resolve(contracts);
-          })
-          .catch(function (response) {
-            deferred.reject(response);
-          });
+      if (_.isEmpty(contracts)) {
```

Comment:
I think this block can be refactored to this:

```js
function getContracts () {
  if (!_.isEmpty(contracts)) {
    return $q.resolve(contracts);
  }

  return getJobContracts().then(filterActiveContracts);
}
```

Reply:
Perfect!!!

Reply:
Nice suggestion!  But we need the following statement to set the `contracts` in the serivce:
`contracts = filterActiveContracts(jobContracts);`

So I  have refactored as follows instead:
```js
/**
 * Returns active job contracts
 *
 * @return {Promise}
 */
 function getContracts () {
    if (!_.isEmpty(contracts)) {
       return $q.resolve(contracts);
    }

    return getJobContracts()
       .then(function (jobContracts) {
          contracts = filterActiveContracts(jobContracts);
          $q.resolve(contracts);
       });
 }
```
What do you say?